Library/Homebrew/os/mac/sdk.rb

```diff
@@ -52,7 +52,17 @@ def sdk_if_applicable(v = nil)
         rescue NoSDKError
           latest_sdk
         end
-        # Only return an SDK older than the OS version if it was specifically requested
+        # Accept an SDK for another OS version if it shares a major version
+        # with the current OS - for example, the 11.0 SDK on 11.1,
+        # or vice versa.
+        # Note that this only applies on macOS 11
+        # or greater, given the way the versioning has changed.
+        # This shortcuts the below check, since we *do* accept an older version
+        # on macOS 11 or greater if the major version matches.
+        return sdk if OS::Mac.version.major >= 11 && sdk.version.major == OS::Mac.version.major
```

Comment:
Can this use the :big_sur symbol rather than using 11? This should also handle the case where sdk is nil.

Reply:
Done!